src/plugins/ui_stateful/clipboard.ts

```diff
@@ -152,10 +155,19 @@ export class ClipboardPlugin extends UIPlugin {
         break;
       }
       case "PASTE_FROM_OS_CLIPBOARD":
-        const state = new ClipboardOsState(cmd.text, this.getters, this.dispatch, this.selection);
-        state.paste(cmd.target);
+        this.state = new ClipboardOsState(cmd.text, this.getters, this.dispatch, this.selection);
+        this.state.paste(cmd.target);
+        this.lastPasteState = this.state;
         this.status = "invisible";
         break;
+      case "REPEAT_PASTE": {
+        this.lastPasteState?.paste(cmd.target, {
+          pasteOption: cmd.pasteOption,
```

Comment:
As pasteOption can be undefined, shouldn't we keep the option of the "last" past ? Maybe it's the case and I simply missed it